pkg/kube/inject/webhook.go

```diff
@@ -216,6 +226,11 @@ func NewWebhook(p WebhookParameters) (*Webhook, error) {
 
 // Run implements the webhook server
 func (wh *Webhook) Run(stop <-chan struct{}) {
+	if wh.namespaces != nil {
+		if !kube.WaitForCacheSync("namespace controller", stop, wh.namespaces.HasSynced) {
```

Comment:
```
// addStartFunc appends a function to be run. These are run synchronously in order,
// so the function should start a go routine if it needs to do anything blocking
```

So we should not be blocking like this. Instead we would need run `wh.Run` as `go wh.Run` and add a HasSynced on webhook that polls and blocks readiness I think.

This is also blocking the wrong thing I think - this blocks the watcher, but the injection server is already started and marked ready at L86 of sidecarinjector.go. Technically I think this would work out since this will block the 'global readiness' but we should be cautious of the webhook server vs webhook patcher difference

Reply:
I've added `HasSynced()` in the webhook and I'm calling it from `server .cachesSynced()`.